sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py

```diff
@@ -716,35 +716,35 @@ def connect(self):
                 WebSocketTimeoutException,
                 WebSocketConnectionClosedException
             )
-
-            self.ws = create_connection(
-                url="wss://{}".format(self._custom_endpoint or self._host),
-                subprotocols=[AMQP_WS_SUBPROTOCOL],
-                timeout=self._connect_timeout,
-                skip_utf8_validation=True,
-                sslopt=self.sslopts,
-                http_proxy_host=http_proxy_host,
-                http_proxy_port=http_proxy_port,
-                http_proxy_auth=http_proxy_auth,
-            )
-        except WebSocketAddressException as exc:
-            raise AuthenticationException(
-                ErrorCondition.ClientError,
-                description="Failed to authenticate the connection due to exception: " + str(exc),
-                error=exc,
-            )
-        # TODO: resolve pylance error when type: ignore is removed below, issue #22051
-        except (WebSocketTimeoutException, SSLError, WebSocketConnectionClosedException) as exc:    # type: ignore
-            self.close()
-            raise ConnectionError("Websocket failed to establish connection: %r" % exc) from exc
-        except (OSError, IOError, SSLError) as e:
-            _LOGGER.info("Websocket connection failed: %r", e, extra=self.network_trace_params)
-            self.close()
-            raise
-        except ImportError:
+            try:
+                self.ws = create_connection(
+                    url="wss://{}".format(self._custom_endpoint or self._host),
+                    subprotocols=[AMQP_WS_SUBPROTOCOL],
+                    timeout=self._connect_timeout,
+                    skip_utf8_validation=True,
+                    sslopt=self.sslopts,
+                    http_proxy_host=http_proxy_host,
+                    http_proxy_port=http_proxy_port,
+                    http_proxy_auth=http_proxy_auth,
+                )
+            except WebSocketAddressException as exc:
+                raise AuthenticationException(
+                    ErrorCondition.ClientError,
+                    description="Failed to authenticate the connection due to exception: " + str(exc),
+                    error=exc,
+                )
+            # TODO: resolve pylance error when type: ignore is removed below, issue #22051
+            except (WebSocketTimeoutException, SSLError, WebSocketConnectionClosedException) as exc:    # type: ignore
+                self.close()
+                raise ConnectionError("Websocket failed to establish connection: %r" % exc) from exc
+            except (OSError, IOError, SSLError) as e:
+                _LOGGER.info("Websocket connection failed: %r", e, extra=self.network_trace_params)
+                self.close()
+                raise
+        except ModuleNotFoundError as exc:
```

Comment:
Curious why this in no longer an `ImportError`?

Reply:
I believe this was always a ModuleNotFoundError, but b/c ModuleNotFoundError is a subclass of ImportError, excepting on ImportError worked.

Reply:
Yeah, I'm curious why we're going with the more specific error rather than the broader ImportError - is there a reason this was causing conflicts?
In this kind of core pattern - I would recommend just wrapping the import statements themselves in the try/except, rather than a whole block of code that is never going to raise an ImportError.

Reply:
No particular reason for switching to the more specific exception. Updated all the ModuleNotFoundErrors to ImportError + only wrapping the try/except ImportError around the import statements.